### PR TITLE
upgrade to latest dependencies

### PR DESCRIPTION
**Pull request by creydr**
Same as #4019 but updates the statefulsetscheduler usage to use the podlister (from https://github.com/knative/eventing/pull/8144)

Got the following error in the kafka-controller otherwise:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1d381c1]

goroutine 694 [running]:
knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup.(*Reconciler).schedule(0xc0007320f0, {0x26fe428, 0xc000e03b90}, 0xc0008a0388)
	knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup/consumergroup.go:452 +0x1e1
knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup.(*Reconciler).ReconcileKind(0xc0007320f0, {0x26fe428, 0xc000e03b90}, 0xc0008a0388)
	knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup/consumergroup.go:201 +0x1cb
knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/injection/reconciler/eventing/v1alpha1/consumergroup.(*reconcilerImpl).Reconcile(0xc0009b01e0, {0x26fe428, 0xc000e03b60}, {0xc000995d10, 0x2a})
	knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/injection/reconciler/eventing/v1alpha1/consumergroup/reconciler.go:240 +0x52b
knative.dev/pkg/controller.(*Impl).processNextWorkItem(0xc00095cc00)
	knative.dev/pkg@v0.0.0-20240815051656-89743d9bbf7c/controller/controller.go:540 +0x508
knative.dev/pkg/controller.(*Impl).RunContext.func3()
	knative.dev/pkg@v0.0.0-20240815051656-89743d9bbf7c/controller/controller.go:489 +0x53
created by knative.dev/pkg/controller.(*Impl).RunContext in goroutine 667
	knative.dev/pkg@v0.0.0-20240815051656-89743d9bbf7c/controller/controller.go:487 +0x2fc
```

### CODE REVIEW

**Comment by creydr** [comment]
@matzew @pierDipi could you PTAL?

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module knative.dev/eventing-kafka-broker
 
-go 1.22
+go 1.22.0
```
Comment (pierDipi) [inline go.mod:3]: why?

Comment (creydr) [reply go.mod:3]: This is coming from update-deps. I asked the same question in [Slack](https://cloud-native.slack.com/archives/C04LMU33V1S/p1723480072792539) and it seems we need to have this for some time: https://cloud-native.slack.com/archives/C04LMU33V1S/p1723547259859939?thread_ts=1723480072.792539&cid=C04LMU33V1S

File: control-plane/pkg/reconciler/consumergroup/controller.go [hunk 2]
```diff
@@ -378,6 +379,7 @@ func createStatefulSetScheduler(ctx context.Context, c SchedulerConfig, lister s
 		Evictor:              newEvictor(ctx, zap.String("kafka.eventing.knative.dev/component", "evictor")).evict,
 		VPodLister:           lister,
 		NodeLister:           nodeinformer.Get(ctx).Lister(),
+		PodLister:            podinformer.Get(ctx).Lister().Pods(system.Namespace()),
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/consumergroup/controller.go:382]: here we need to pass the `dispatcherPodInformer`

Comment (creydr) [reply control-plane/pkg/reconciler/consumergroup/controller.go]: :+1:

**Comment by creydr** [comment]
@pierDipi can you recheck?

**Pull request merged**